### PR TITLE
[stable/fluent-bit] Add existing secret for ES credentials

### PR DESCRIPTION
**Pull request by endrec**
#### What this PR does / why we need it:
This changest adds the possibility to use an existing K8S secret for ES credentials and certificate.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by kfox1111** [comment]
lgtm

**Comment by edsiper** [comment]
Looks like chart version is not correct.

**Comment by endrec** [comment]
@edsiper Corrected version. It was correct, when I filed the PR. :stuck_out_tongue:

File: stable/fluent-bit/templates/daemonset.yaml [hunk 1]
```diff
@@ -33,7 +33,16 @@ spec:
...
         imagePullPolicy: "{{ .Values.image.pullPolicy }}"
         env:
+{{- if .Values.backend.es.http_passwd_secret }}
+          - name: HTTP_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: {{ .Values.backend.es.http_passwd_secret }}
+                key: {{ .Values.backend.es.http_passwd_secret_key }}
+{{- end }}
+{{- if or .Values.env (not .Values.backend.es.http_passwd_secret) }}
```
Comment (kfox1111) [inline stable/fluent-bit/templates/daemonset.yaml:43]: I'm not following the logic here...

If values is nothing, and http_password_secret is nothing, still toYaml .Values.env?

Shouldn't this be this:
{{- if .Values.env }}

?

File: stable/fluent-bit/templates/daemonset.yaml [hunk 2]
```diff
@@ -78,11 +87,10 @@ spec:
           mountPath: /fluent-bit/etc/parsers_custom.conf
           subPath: parsers.conf
 {{- end }}
-{{- end }}
```
Comment (kfox1111) [inline stable/fluent-bit/templates/daemonset.yaml:81]: This seems broken here... where did the end go? its merging a large chunk of code below into the above 'else' incorrectly?

File: stable/fluent-bit/templates/daemonset.yaml [hunk 3]
```diff
@@ -112,7 +120,11 @@ spec:
...
           path: /var/lib/docker/containers
-{{- if .Values.backend.es.tls_ca }}
+{{- if .Values.backend.es.tls_secret }}
+      - name: es-tls-secret
+        secret:
+          secretName: {{ .Values.backend.es.tls_secret }}
+{{- else if .Values.backend.es.tls_ca }}
       - name: es-tls-secret
         secret:
           secretName: "{{ template "fluent-bit.fullname" . }}-es-tls-secret"
```
Comment (kfox1111) [inline stable/fluent-bit/templates/daemonset.yaml:130]: Could you just put the conditional around the secretName line? Everything else seems to be the same.

**Review by kfox1111 — CHANGES_REQUESTED** [review]
A few issues

**Pull request closed without merging**